tool/data/generator/oo.loader.tmpl.js

```diff
@@ -2,11 +2,12 @@
 
 if (!this.window) window = this;
 
-if (!window.navigator) window.navigator = {
-  userAgent: "Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_6_4; de-de) AppleWebKit/533.17.8 (KHTML, like Gecko) Version/5.0.1 Safari/533.17.8", 
-  product: "", 
-  cpuClass: ""
-}; 
+if (!window.navigator) window.navigator = {};
+if (!window.navigator.userAgent) {
+  window.navigator.userAgent = "Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_6_4; de-de) AppleWebKit/533.17.8 (KHTML, like Gecko) Version/5.0.1 Safari/533.17.8";
```

Comment:
since this 'bug' is about react-native, how about actually detecting if we are running under react native?

others have run into this problem too it seems https://github.com/facebook/react-native/issues/1331

https://github.com/facebook/react-native/blob/da3424c929220304ff01032a9bf473fb251375f1/Libraries/Core/InitializeCore.js#L148

so with

`global.navigator.product === 'ReactNative'`

Reply:
@oetiker I think this PR is the right way for a generic initialization. From what I read under react `window.navigator.product` should have the value `ReactNative` as it will be set by react itself.
So this PR initializes all questionable attributes if not already set by an arbitrary runtime.

Reply:
Or do you mean that we should initialize `window.navigator.userAgent` with some value containing the string `ReactNative`?

Reply:
It seems bad form to assume that if an object exists, all of its members that exist in some environments will always exist in others as well. I think the change here is actually the best way of ensuring, in the general case, that all expected members now exist, without making any assumptions about what existed initially. We could *then* check for specific environments and do environment-specific things if necessary. (I don't see any being necessary in this case, though.)

Reply:
I'd argue that this (ie the current PR implementation) is a more flexible "feature detection"-style approach instead of detecting an actual platform

Reply:
`window && window.process && window.process.type` can detect if we're running in *electron*. Maybe adding some generic env info about 'where am I running in' is helpful.

Reply:
what irks me is that we set an actual user agent string which might cause confusion further down the line ...

Reply:
but yes this is not a new features ... so ... I guess the PR is ok